### ISSUE TITLE
[Collaborative Filtering] Armadillo matrix initialization error

### ISSUE THREAD

**Issue by DragonOnTheBeach** [mlpack/mlpack#857]
Hi,

I tried executing collaborative filtering sample program from the link:
http://www.mlpack.org/docs/mlpack-2.0.2/doxygen.php?doc=cftutorial.html

* CF with default parameters

The code used is-
```
#include <mlpack/methods/cf/cf.hpp>
#include <mlpack/core/data/load.hpp>
using namespace mlpack::cf;

int main(){

std::string inputFile = "
[dataset_small.csv.zip](https://github.com/mlpack/mlpack/files/725918/dataset_small.csv.zip)

dataset_small.csv";

arma::mat data;
size_t neighborhood;
size_t rank;
mlpack::data::Load(inputFile ,data ,true);
// Build the CF object and perform the decomposition.
// The constructor takes a default-constructed factorizer, which, by default,
// is of type amf::NMFALSFactorizer.
std::cout << neighborhood << '\n';
std::cout << rank << '\n';
std::cout << data << '\n';

CF cf(data, mlpack::amf::NMFALSFactorizer(), neighborhood, rank);
std::cout << "data_1" << '\n';
arma::Mat<size_t> recommendations;
cf.GetRecommendations(5, recommendations);
}
```

Compilation has been done through-
```
g++ -std=c++11 collaborative_filtering_mlpack.cpp -larmadillo -lmlpack
```

Compilation happened successfully, after that, while trying to run the same, the error I got is-
```
140423955742844
725871085
   1.9600e+02   1.8600e+02   2.2000e+01   2.4400e+02   1.6600e+02   1.1500e+02
   2.4200e+02   3.0200e+02   3.7700e+02   5.1000e+01   3.4600e+02   2.6500e+02
   3.0000e+00   3.0000e+00   1.0000e+00   2.0000e+00   1.0000e+00   2.0000e+00


error: Mat::init(): requested size is too large

terminate called after throwing an instance of 'std::logic_error'
  what():  Mat::init(): requested size is too large
Aborted (core dumped)
```

The file used as an input is attached.

I ran the mlpack collaborative filtering on the same file using 
```
mlpack_cf -A -t dataset_small.csv -v -o recommendations.csv
```
and it worked fine.

The error is only while running the compiled code. Request you to provide a solution.

**Comment by rcurtin** [comment]
The variable `rank` isn't initialized---make sure you set that to what you want the rank of the decomposition to be, and it should work.  Since `rank` is left unset, it will probably have a random value that might be very large (or it might be 0, which will make the program fail in a different way). :)

**Comment by mohammadAdnan** [comment]
rcurtin, how are you? My first steps in C++ with mlpack is little slow. After i solved my last issue regarding building mlpack, now i am stuck trying to run my first program. I tried out NeighborSearch tutorial and collaborative filtering using visual studio 2015. For NeighborSearch, I moved armadillo folder and mlpack in order to include them in my project. I then added the blas.lib and lapack.lib to project dependencies. But i am still receiving errors: https://mega.nz/#!9VwQAZ4Z!m9VFfqTuiDb2rCeKtdvNQ7qYyMOX7gcjsAa9B8o9W1E
